spec.bs

```diff
@@ -394,21 +366,27 @@ Moreover, each {{SharedStorageWorklet}}'s [=global scopes|list of global scopes=
                 :   If it was fulfilled with value |index|:
                 ::  1. If |index| is greater than |urlList|'s [=list/size=], then:
                         1. If |savedQueryName| is a [=string=] that is not the empty string, then run [=store the index for a saved query=] with |window|, |navigable|, |workletDataOrigin|, |moduleURLRecord|, |operationName|, |savedQueryName|, and the [=default selectURL index=].
-                        1. [=Queue a global task=] on the [=DOM manipulation task source=], given |window|, to [=reject=] |promise| with a {{TypeError}}, and abort these steps.
+                        1. [=Queue a global task=] on the [=DOM manipulation task source=], given |window|, to [=reject=] |promise| with a {{TypeError}}.
+                        1. Run |privateAggregationCompletionTask| given [=Private Aggregation completion task trigger/normal completion=].
+                        1. Return (|promise|, false).
 
                         Note: The result index is beyond the input urls' size. This violates the selectURL() protocol, and we don't know which url should be selected.
 
-                        Otherwise:
+                    1. Otherwise:
                         1. If |savedQueryName| is a [=string=] that is not the empty string, then run [=store the index for a saved query=] with |window|, |navigable|, |workletDataOrigin|, |moduleURLRecord|, |operationName|, |savedQueryName|, and |index|.
                         1. [=Queue a global task=] on the [=DOM manipulation task source=], given |window|, to [=resolve=] |promise| with |index|.
-                        1. Run |privateAggregationCompletionTask|.
+                        1. Run |privateAggregationCompletionTask| given [=Private Aggregation completion task trigger/normal completion=].
 
                 :   If it was rejected:
                 ::  1. If |savedQueryName| is a [=string=] that is not the empty string, then run [=store the index for a saved query=] with |window|, |navigable|, |workletDataOrigin|, |moduleURLRecord|, |operationName|, |savedQueryName|, and the [=default selectURL index=].
                     1. [=Queue a global task=] on the [=DOM manipulation task source=], given |window|, to [=reject=] |promise| with a {{TypeError}}.
 
                         Note: This indicates that either |operationCtor|'s run() method encounters an error (where |operationCtor| is the parameter in {{SharedStorageWorkletGlobalScope/register()}}), or the result |index| is a non-integer value, which violates the selectURL() protocol, and we don't know which url should be selected.
-                    1. Run |privateAggregationCompletionTask|.
+                    1. If the promise was rejected as the operation was completed abruptly due to an uncaught exception:
```

Comment:
Wasn't sure how to evaluate this properly (in spec language) unless we decide that returning a non-integer should count as an uncaught exception for aggregate error reporting

Reply:
A non-integer will often be implicitly converted to an integer without error, see test case [SelectURL_NonNumericStringConvertedTo0](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/shared_storage/shared_storage_worklet_unittest.cc?q=SelectURL_NonNumericStringConvertedTo0). AFAICT, the only possible integer conversion failure is to return an object that defines a custom toString() method that throws, e.g. [SelectURL_ReturnValueToUint32Error](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/shared_storage/shared_storage_worklet_unittest.cc?q=SelectURL_ReturnValueToUint32Error). I think it's fine to simply assume 'uncaught exception' here.

Reply:
Ah interesting -- I've updated this PR to reflect that / remove the unnecessary if clause. I'll try to add a test case for this situation in the implementation too. Thanks!